src/main/drivers/at32/bus_spi_at32bsp.c

```diff
@@ -324,67 +327,73 @@ void spiSequenceStart(const extDevice_t *dev)
         bus->busType_u.spi.leadingEdge = dev->busType_u.spi.leadingEdge;
     }
 
-    spi_enable(instance,TRUE);
+    spi_enable(instance, TRUE);
 
-    // Check that any there are no attempts to DMA to/from CCD SRAM
+    // Count segments
     for (busSegment_t *checkSegment = (busSegment_t *)bus->curSegment; checkSegment->len; checkSegment++) {
-        // Check there is no receive data as only transmit DMA is available
-        if (((checkSegment->u.buffers.rxData) && (IS_CCM(checkSegment->u.buffers.rxData) || 
-             (bus->dmaRx == (dmaChannelDescriptor_t *)NULL))) ||
-            ((checkSegment->u.buffers.txData) && IS_CCM(checkSegment->u.buffers.txData))) {
-            dmaSafe = false;
-            break;
-        }
         // Note that these counts are only valid if dmaSafe is true
         segmentCount++;
         xferLen += checkSegment->len;
     }
+
     // Use DMA if possible
-    // If there are more than one segments, or a single segment with negateCS negated then force DMA irrespective of length
-    if (bus->useDMA && dmaSafe && ((segmentCount > 1) || (xferLen >= 8) || !bus->curSegment->negateCS)) {
+    // If there are more than one segments, or a single segment with negateCS negated in the list terminator then force DMA irrespective of length
+    if (bus->useDMA && dmaSafe && ((segmentCount > 1) ||
+                                   (xferLen >= SPI_DMA_THRESHOLD) ||
+                                   !bus->curSegment[segmentCount].negateCS)) {
+        // Intialise the init structures for the first transfer
         spiInternalInitStream(dev, false);
+
+        // Assert Chip Select
         IOLo(dev->busType_u.spi.csnPin);
+
+        // Start the transfers
         spiInternalStartDMA(dev);
     } else {
         busSegment_t *lastSegment = NULL;
+        bool segmentComplete;
 
+        // Manually work through the segment list performing a transfer for each
         while (bus->curSegment->len) {
             if (!lastSegment || lastSegment->negateCS) {
+                // Assert Chip Select if necessary - it's costly so only do so if necessary
                 IOLo(dev->busType_u.spi.csnPin);
             }
 
             spiInternalReadWriteBufPolled(
                     bus->busType_u.spi.instance,
                     bus->curSegment->u.buffers.txData,
                     bus->curSegment->u.buffers.rxData,
-                    bus->curSegment->len
-                );
+                    bus->curSegment->len);
```

Comment:
Indentation ;)

Reply:
Hopefully you're happier with how it looks now.